### PR TITLE
feat(sdk/go): Add SIP-002 "Application Configuration" to TinyGo SDK

### PR DESCRIPTION
**Pull request by adamreese**

### CODE REVIEW

File: examples/config-tinygo/readme.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+content-type: text/plain; charset=utf-8
+content-length: 16
+date: Wed, 09 Mar 2022 17:23:08 GMT
+
+Hello, Fermyon!
+```
+
+## Notes
+
+- components built using TinyGo will be run in Spin using the Wagi executor
```
Comment (radu-matei) [inline examples/config-tinygo/readme.md:74]: I don't think this readme is accurate anymore.

File: examples/config-tinygo/main.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	spinconfig "github.com/fermyon/spin/sdk/go/config"
```
Comment (radu-matei) [inline examples/config-tinygo/main.go:7]: I'm torn between prefixing `spin*` before packages everywhere, or just where it clashes with standard library packages.
(nothing to change here, just wondering what your thoughts are)

File: examples/config-tinygo/main.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	spinhttp "github.com/fermyon/spin/sdk/go/http"
+)
+
+func init() {
+	spinhttp.Handle(func(w http.ResponseWriter, r *http.Request) {
+		if val, err := spinconfig.Get("message"); err != nil {
```
Comment (radu-matei) [inline examples/config-tinygo/main.go:13]: Would you mind adding a comment here describing where the value is coming from?
Thanks!

File: examples/config-tinygo/readme.md (hunk 1 above)
Comment (adamreese) [reply examples/config-tinygo/readme.md:74]: copy-pasta

File: examples/config-tinygo/main.go (hunk 2 above)
Comment (adamreese) [reply examples/config-tinygo/main.go:7]: I think only where it clashes.

**Review by radu-matei — APPROVED** [review]
Thanks!

LGTM

**Pull request merged**